pennylane/circuit_graph.py

```diff
@@ -206,6 +211,124 @@ def hash(self):
         """
         return hash(self.serialize())
 
+    def to_openqasm(self, rotations=True):
+        """Serialize the circuit as an OpenQASM 2.0 program.
+
+        Only operations are serialized; all measurements
+        are assumed to take place in the computational basis.
+
+        .. note::
+
+            The serialized OpenQASM program assumes that gate definitions
+            in ``qelib1.inc`` are available.
+
+        Args:
+            rotations (bool): in addition to serializing user-specified
+                operations, also include the gates that diagonalize the
+                measured wires such that they are in the eigenbasis of the circuit observables.
+
+        Returns:
+            str: OpenQASM serialization of the circuit
+        """
+        # We import decompose_queue here to avoid a circular import
+        from pennylane.qnodes.base import decompose_queue  # pylint: disable=import-outside-toplevel
```

Comment:
How about having the [importing the entire module](https://stackoverflow.com/a/22210807/12899253) `import pennylane.qnodes.base`?

Reply:
I'll give it a go and see if it works!

Reply:
Unfortunately it still causes a circular import bug. We should consider moving the `decompose` function out of the `pennylane.qnodes.base` module into it's own module, but that is out of scope for this PR.